exporters/stats/prometheus/src/main/java/io/opencensus/exporter/stats/prometheus/PrometheusExportUtils.java

```diff
@@ -0,0 +1,208 @@
+import com.google.common.collect.Lists;
+import io.opencensus.common.Function;
+import io.opencensus.common.Functions;
+import io.opencensus.stats.Aggregation;
+import io.opencensus.stats.Aggregation.Count;
+import io.opencensus.stats.Aggregation.Distribution;
+import io.opencensus.stats.Aggregation.Mean;
+import io.opencensus.stats.Aggregation.Sum;
+import io.opencensus.stats.AggregationData;
+import io.opencensus.stats.AggregationData.CountData;
+import io.opencensus.stats.AggregationData.DistributionData;
+import io.opencensus.stats.AggregationData.MeanData;
+import io.opencensus.stats.AggregationData.SumDataDouble;
+import io.opencensus.stats.AggregationData.SumDataLong;
+import io.opencensus.stats.View;
+import io.opencensus.stats.View.AggregationWindow;
+import io.opencensus.stats.View.AggregationWindow.Cumulative;
+import io.opencensus.stats.ViewData;
+import io.opencensus.tags.TagKey;
+import io.opencensus.tags.TagValue;
+import io.prometheus.client.Collector;
+import io.prometheus.client.Collector.MetricFamilySamples;
+import io.prometheus.client.Collector.MetricFamilySamples.Sample;
+import io.prometheus.client.Collector.Type;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map.Entry;
+
+/*>>>
+import org.checkerframework.checker.nullness.qual.Nullable;
+*/
+
+/**
+ * Util methods to convert OpenCensus Stats data models to Prometheus data models.
+ *
+ * <p>Each OpenCensus {@link View} will be converted to a Prometheus {@link MetricFamilySamples}
+ * with no {@link Sample}s, and is used for registering Prometheus {@code Metric}s. Only {@link
+ * Cumulative} views are supported. All views are under namespace "opencensus".
+ *
+ * <p>{@link Aggregation} will be converted to a corresponding Prometheus {@link Type}. {@link Sum}
+ * will be {@link Type#UNTYPED}, {@link Count} will be {@link Type#COUNTER}, {@link Mean} will be
+ * {@link Type#SUMMARY} and {@link Distribution} will be {@link Type#HISTOGRAM}. Please note we
+ * cannot set bucket boundaries for custom {@link Type#HISTOGRAM}.
+ *
+ * <p>Each OpenCensus {@link ViewData} will be converted to a Prometheus {@link
+ * MetricFamilySamples}, and each {@code Row} of the {@link ViewData} will be converted to
+ * Prometheus {@link Sample}s.
+ *
+ * <p>{@link SumDataDouble}, {@link SumDataLong} and {@link CountData} will be converted to a single
+ * {@link Sample}. {@link MeanData} will be converted to two {@link Sample}s sum and count. {@link
+ * DistributionData} will be converted to a list of {@link Sample}s that have the sum, count and
+ * histogram buckets.
+ *
+ * <p>{@link TagKey} and {@link TagValue} will be converted to Prometheus {@code LabelName} and
+ * {@code LabelValue}. {@code Null} {@link TagValue} will be converted to an empty string.
+ *
+ * <p>Please note that Prometheus Metric and Label name can only have alphanumeric characters and
+ * underscore. All other characters will be sanitized by underscores.
```

Comment:
This seems reasonable and consistent with what we've done in the past but it has also lead to some confusion in the past too.
1. Is this what the Census Go exporter does?
2. Does anyone have any concerns about this?

Reply:
One more question:
How do we handle the case of duplicated entries after sanitization:
"grpc/io/test" -> "grpc_io_test"
"grpc.io/test" -> "grpc_io_test"

In this case I think we need to do a merge of the values.

Reply:
Yes, this is consistent with the current Go implementation: https://github.com/census-instrumentation/opencensus-go/blob/master/exporter/prometheus/prometheus.go#L276.
@jcd2 had some concerns about the subtle corner cases in https://github.com/census-instrumentation/opencensus-go/issues/173, including the duplicated names after sanitization. 

In Java we're using the [standard sanitize method provided by Prometheus](https://github.com/prometheus/client_java/blob/master/simpleclient/src/main/java/io/prometheus/client/Collector.java#L181) and having the Prometheus client deal with the corner cases, just like any other direct users of Prometheus. For example, for the case of duplicated entries after sanitization, [Prometheus client will simply reject the second entry](https://github.com/prometheus/client_java/blob/master/simpleclient/src/main/java/io/prometheus/client/CollectorRegistry.java#L54).

I'm not sure if we should wrap around such cases in our exporter though (like merging the values). To me it's like a hack around Prometheus restrictions, and might lead to confusions to the users. It might be better to just follow the Prometheus client library convention on these cases.

Reply:
Discussed with @dinooliva and @ramonza , we decided to keep consistent with Go for the moment. I've created https://github.com/census-instrumentation/opencensus-specs/issues/50 to follow up on this.